### DISCUSSION TITLE
I am Trying to upload file to s3 using boto3 library using upload_fileobj function in FastAPI able to track progress of upload using callback function how to send these  realtime progress updates to UI

### DISCUSSION THREAD

**Discussion by mounikabonam** [fastapi/fastapi#10859] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, File, UploadFile, HTTPException, Form
import boto3
app = FastAPI()
s3 = boto3.client('s3')
BUCKET_NAME = 'your_bucketname'




def upload_progress_callback(total_size, bytes_transferred,total_bytes_transferred):
    total_bytes_transferred[0] += bytes_transferred
    print(total_bytes_transferred)
    percentage = (total_bytes_transferred[0] / total_size) * 100
    print(f"Progress: {percentage:.2f}%")




@app.post("/uploadfile/obj/")
async def create_upload_file_using_uploadfileobj(file: UploadFile = File(...)):
    #s3.upload_fileobj(Bucket=BUCKET_NAME, Key=file.filename, Body=await file.read())
    total_size= file.size
    total_bytes_transferred = [0]
    s3.upload_fileobj(file.file, BUCKET_NAME, file.filename,Callback=lambda bytes_transferred: upload_progress_callback(total_size, bytes_transferred, total_bytes_transferred))
    return {"filename": file.filename}



if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="127.0.0.1", port=8000)
```

### Description

I am trying to send realtime progress of fileuplaod. Everywhere iam finding like useof websockets will help this issue but unable to understand how to implement in this case

### Operating System

Windows

### FastAPI Version

0.108.0

### Pydantic Version

2.5.3

### Python Version

Python 3.10.10

**Comment by aanchlia** [comment]
Can you try this to see if this works, very simplified version.

```python
def upload_to_s3(file, object_name, callback):
    s3.upload_fileobj(
        Fileobj=file.file,
        Bucket=bucket_name,
        Key=object_name,
        Callback=callback
    )

def progress_callback(bytes_transferred):
    print(f"Uploaded {bytes_transferred} bytes")

@app.post("/upload/")
async def upload_file(file: UploadFile = File(...)):
    upload_to_s3(file, file.filename, progress_callback)
    return {"message": "Upload successful"}
```

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
